semantic/lookup_test.go

```diff
@@ -0,0 +1,163 @@
+	semantic "github.com/influxdata/flux/semantic"
+)
+
+func TestLookupSimpleTypes(t *testing.T) {
+	for _, testCase := range []struct {
+		path string
+		id   string
+		name string
+		want string
+	}{
+		{path: "math", id: "pi", name: "lookup math.pi", want: "float"},
+		{path: "math", id: "maxint", name: "lookup math.maxint", want: "int"},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			got, _ := semantic.LookupBuiltInType(testCase.path, testCase.id)
+			if want, got := testCase.want, got.String(); want != got {
+				t.Fatalf("unexpected result -want/+got\n\t- %s\n\t+ %s", want, got)
+			}
+		})
+	}
+}
+
+func TestLookupComplexTypes(t *testing.T) {
+	for _, testCase := range []struct {
+		path string
+		id   string
+		name string
+		want string
+	}{
+		{
+			path: "csv",
+			id:   "from",
+			name: "lookup csv.from",
+			want: "(?csv: string, ?file: string) -> [t0]",
+		},
+		{
+			path: "date",
+			id:   "nanosecond",
+			name: "lookup date.nanosecond",
+			want: "(t: time) -> int", 
+		}, 
+		{
+			path: "date",
+			id:   "truncate",
+			name: "lookup date.truncate",
+			want: "(t: time, unit: duration) -> time", 
+		},
+		{
+			path: "experimental/bigtable",
+			id:   "from",
+			name: "lookup experimental/bigtable.from",
+			want: "(instance: string, project: string, table: string, token: string) -> [t0]", 
+		},
+		{
+			path: "experimental",
+			id:   "to",
+			name: "lookup experimental.to",
+			want: "(?bucket: string, ?bucketID: string, ?host: string, ?org: string, ?orgID: string, <-tables: [t0], ?token: string) -> [t0]", 
+		},
+		{
+			path: "http",
+			id:   "post",
+			name: "lookup http.post",
+			want: "(?data: bytes, ?headers: t0, url: string) -> int", 
+		},
+		{
+			path: "influxdata/influxdb/secrets",
+			id:   "get",
+			name: "lookup influxdata/influxdb/secrets.get",
+			want: "(key: string) -> string", 
+		},
+		{
+			path: "json",
+			id:   "encode",
+			name: "lookup json.encode",
+			want: "(v: t0) -> bytes", 
+		},
+		{
+			path: "strings",
+			id:   "title",
+			name: "lookup strings.title",
+			want: "(v: string) -> string", 
+		},
+		{
+			path: "system",
+			id:   "time",
+			name: "lookup system.time",
+			want: "() -> time", 
+		},
+		{
+			path: "universe",
+			id:   "bool",
+			name: "lookup universe.bool",
+			want: "(v: t0) -> bool",
+		},
+		{
+			path: "internal/promql",
+			id:   "changes",
+			name: "lookup internal/promql.changes",
+			want: "(<-tables: [{_value: float | t0}]) -> [{_value: float | t1}]", 
+		},
+		{
+			path: "sql",
+			id:   "to",
+			name: "lookup sql.to",
+			want: "(?batchSize: int, dataSourceName: string, driverName: string, table: string, <-tables: [t0]) -> [t0]",
+		},
+		{
+			path: "testing",
+			id:   "assertEmpty",
+			name: "lookup testing.assertEmpty",
+			want: "(<-tables: [t0]) -> [t0]",
+		},
+		{
+			path: "universe",
+			id:   "chandeMomentumOscillator",
+			name: "lookup testing.assertEmpty",
+			want: "(?columns: [string], n: int, <-tables: [t0]) -> [t1]",
+		},
```

Comment:
This testing looks good.  I don't think you need to lock down every single builtin here.  Personally I think it's enough to just assert that a few of the builtins are here so you know that lookup is working.

Reply:
Are there too many test cases here? Should I remove some? My intention was to provide coverage of every stdlib package and various PolyType representations.

Reply:
I think what you have here is fine.  

I don't think someone should need to update this test whenever we add something to the stdlib.  It would just be more boilerplate, and we already have roundtrip testing for types.  There will also be tests added to test the functionality of what is added to the stdlib.  If this tests looks like it might be an exhaustive list of all the builtins, it would be confusing if things are missing.

Reply:
I messaged you about this: I'm gonna keep most of these test cases, but clarify that this list is not exhaustive in a comment.